lti_consumer/static/js/xblock_lti_consumer.js

```diff
@@ -92,68 +105,97 @@ function LtiConsumerXBlock(runtime, element) {
+            // Hide the button that triggered the event, i.e. the launch button.
+            triggerElement.hide();
+
+            $('<div id="dialog-container"></div>').insertAfter(triggerElement) // TODO: this will need some cute styling. It looks like trash but it works.
+                .append('<p>' + message + '</p>')
+            if (showCancelButton) {
+                $('#dialog-container')
+                .append('<button style="margin-right:1rem" id="cancel-button">Cancel</button>');
+            }
+            $('#dialog-container').append('<button id="confirm-button">OK</button>');
+
+            // When a learner clicks "OK" or "Cancel" in the consent dialog, remove the consent dialog, show the launch
+            // button, and resolve the promise.
+            $('#confirm-button').click(function () {
+                // Show the button that triggered the event, i.e. the launch button.
+                triggerElement.show();
+                $("#dialog-container").remove()
+                def.resolve("OK");
+            })
+            $('#cancel-button').click(function () {
+                // Hide the button that triggered the event, i.e. the launch button.
+                triggerElement.show()
+                $("#dialog-container").remove()
+                def.resolve("Cancel");
+            })
+            return def.promise();
+        };
+
         var $element = $(element);
         var $ltiContainer = $element.find('.lti-consumer-container');
         var askToSendUsername = $ltiContainer.data('ask-to-send-username') == 'True';
         var askToSendEmail = $ltiContainer.data('ask-to-send-email') == 'True';
+        var ltiVersion = $ltiContainer.data('lti-version');
 
-        // Apply click handler to modal launch button
-        $element.find('.btn-lti-modal').iframeModal({top: 200, closeButton: '.close-modal'});
-
-        // Apply click handler to new window launch button
-        $element.find('.btn-lti-new-window').click(function(){
-
-            // If this instance is configured to require username and/or email, ask user if it is okay to send them
-            // Do not launch if it is not okay
-            var destination = $(this).data('target')
-
-            function confirmDialog(message) {
-                var def = $.Deferred();
-                $('<div></div>').appendTo('body') // TODO: this will need some cute styling. It looks like trash but it works.
-                  .html('<div><p>' + message + '</p></div>')
-                  .dialog({
-                    modal: true,
-                    title: 'Confirm',
-                    zIndex: 10000,
-                    autoOpen: true,
-                    width: 'auto',
-                    resizable: false,
-                    dialogClass: 'confirm-dialog',
-                    buttons: {
-                      OK: function() {
-                        $('body').append('<h1>Confirm Dialog Result: <i>Yes</i></h1>');
```

Comment:
I don't know what the purpose of these headers was. Anyone feel strongly I should leave them in? They don't appear in the LMS, but I could see them at the bottom of the page in Studio.

Reply:
I was wondering if it was an accessibility thing. Thanks for finding that. I'll add it back in.